codex/contracts/market.nim

```diff
@@ -196,10 +198,18 @@ method subscribeSlotFreed*(market: OnChainMarket,
                            callback: OnSlotFreed):
                           Future[MarketSubscription] {.async.} =
   proc onEvent(event: SlotFreed) {.upraises:[].} =
-    callback(event.slotId)
+    callback(event.requestId, event.slotIndex)
   let subscription = await market.contract.subscribe(SlotFreed, onEvent)
   return OnChainMarketSubscription(eventSubscription: subscription)
 
+method subscribeFulfillment(market: OnChainMarket,
```

Comment:
For every single request added to the slot queue (which could be all requests if we remove the first availability check), then that means we'd need to subscribe to all the contract events for every single request added in the queue, as well as unsubscribe from all of them once removed. This could quickly get out of hand in terms of memory consumption of Subscriptions as well as issues with the number of eventemitter subscriptions allowed by the node.

My initial thought is that we should avoid the easier problem that we know will exist first, then if we run into scalability issues down the road, we could work out a way to scale better given the information we have at the time.